.azure-pipelines.yml

```diff
@@ -0,0 +1,206 @@
+trigger: none # Do not build branches on push.
+pr:
+- master
+
+schedules:
+- cron: "0 0 * * *"
+  displayName: Nightly build
+  branches:
+    include:
+    - master
+  always: true # Also build if there were no code changes, because many of our tests rely on external infrastructure.
+
+variables:
+  BOWER_VERSION: '1.8.8'
+  BUNDLER_VERSION: '2.1.4'
+  COMPOSER_VERSION: '5.1.0' # The version refers to the installer, not to Composer.
+  CONAN_VERSION: '1.18.0'
+  FLUTTER_VERSION: 'v1.12.13+hotfix.9-stable'
+  GO_DEP_VERSION: '0.5.4'
+  PYTHON_PIPENV_VERSION: '2018.11.26'
+  RUST_VERSION: '1.35.0'
+  STACK_VERSION: '2.1.3.20190715'
+  VIRTUALENV_VERSION: '20.0.14'
+
+stages:
+- stage: build_test
+  displayName: Build and Test
+  jobs:
+  # TODO: Consider running the Docker job only if the Dockerfile was changed.
+  - job: 'Docker'
+    pool:
+      vmImage: 'ubuntu-18.04'
+    steps:
+    - task: Docker@2
+      inputs:
+        command: 'build'
+        Dockerfile: 'Dockerfile'
+        tags: |
+          ort
+    # TODO: Add a step to run ORT in the Docker container, e.g. the requirements task, to verify that the Docker
+    #       container works.
+
+  # TODO: Add a separate job to run detekt to prevent running the matrix build if there are detekt errors?
+  #       Downside is that this requires separate cloning. Alternative is to run detekt only on one OS, or to run detekt
+  #       in a separate parallel job for early feedback, and disable detekt in the other jobs.
+  - job: 'Gradle'
+    strategy:
+      matrix:
+        Linux:
+          imageName: 'ubuntu-18.04'
+          gradleWrapper: 'gradlew'
```

Comment:
Do you really need to set this, as 'gradlew' is the default and the '.bat' extension can be omitted on Windows?

Reply:
According to the documentation, yes: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/gradle?view=azure-devops#arguments

Reply:
But those docs also say `Default value: gradlew`. I'm pretty confident this can be dropped.

Reply:
I'd like to try this outside of this PR, I don't want to do any more experiments on this branch, because work on this already took far longer than planned.